packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -147,6 +151,22 @@ const ProductTile = (props) => {
         })
     }, [product, selectableAttributeId, selectableAttributeValue])
 
+    // Retrive product badges
+    let filteredLabels = new Map()
+    if (product?.representedProduct) {
+        // validate the provided badge labels against the product custom properties. This will allow users to use any boolean custom properties as badges
+        filteredLabels = badgeLabels.reduce((map, item) => {
+            if (
+                item.propertyName &&
+                typeof product.representedProduct[item.propertyName] === 'boolean' &&
+                product.representedProduct[item.propertyName] === true
+            ) {
+                map.set(intl.formatMessage(item.label), item.color)
+            }
+            return map
+        }, filteredLabels)
```

Comment:
Minor: I think in this case, it's more readable if we use `forEach` instead. We're not really calculating from all the array items; we're simply saving them into a map.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#when_to_not_use_reduce

```suggestion
        badgeLabels.forEach((item) => {
            if (
                item.propertyName &&
                typeof product.representedProduct[item.propertyName] === 'boolean' &&
                product.representedProduct[item.propertyName] === true
            ) {
                filteredLabels.set(intl.formatMessage(item.label), item.color)
            }
        })
```